spec/forms/submission_form_spec.rb

```diff
@@ -0,0 +1,195 @@
+require 'rails_helper'
+
+RSpec.describe SubmissionForm do
+  let(:contact_method) { create :contact_method_email }
+  let(:location_type)  { create :location_type }
+  let(:service_area)   { create :service_area }
+
+  describe 'creating a new submission' do
+    let(:params) {{
+      form_name: 'Offer_form',
+      privacy_level_requested: 'anyone',
+      service_area: service_area.id,
+      listing_attributes: {
+        type: 'Offer',
+        tag_list: ['errands','groceries'],
+        description: 'on a quiet day i can hear her breathing',
+      },
+      location_attributes: {
+        id: nil,
+        city: 'Anakwashtank',
+        state: 'DC', # statehood now!
+        location_type: location_type.id
+      },
+      person_attributes: {
+        id: nil,
+        preferred_contact_method: contact_method.id,
+        email: 'we@together.coop',
+        name: 'Harriet Tubman',
+      },
+    }}
+
+    subject(:submission) { SubmissionForm.build params }
+
+
+    it 'builds a Submission instance' do
+      expect(submission).to be_a Submission
+    end
+
+    it 'populates submission fields' do
+      expect(submission.privacy_level_requested).to eq 'anyone'
+      expect(submission.form_name).to eq 'Offer_form'
+    end
+
+    it 'sets the specified service area' do
+      expect(submission.service_area).to eq service_area
+    end
+
+    it 'captures whole submission json'
+
+    describe 'has_one person' do
```

Comment:
should this person and location test be under the json section only? perhaps under both?

Reply:
yup re in both -- json is absolute record of initial submission, assns help us run reports later, and navigate (and at some point, possibly manage) collections of listings.
 
and, sounds great